sentinel/src/lib/history.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Utility to track duplicate keys in a certain period of time (periodMins).
+ * Removes expired keys after threshold is reached (purgeThreshold).
+ **/
+const moment = require('moment');
+
+let records = {};
+
+const now = () => {
+  return moment(new Date());
+};
+
+const expired = (key) => {
+  const value = records[key];
+  if(value && value.timestamp) {
+    // Checks difference with history duration in minutes
+    return (now().diff(value.timestamp))/60000 >= module.exports.periodMins;
+  }
+  return true;
+};
+
+const size = () => {
+  return Object.keys(records).length;
+};
+
+const purge = (force=false) => {
+  if(force || size() >= module.exports.purgeThreshold) {
```

Comment:
I think it makes sense to always purge. I assume you're doing this so purge happens less often, but if a record has expired there's no point keeping it in memory and checking it every time. It simplifies the code and reduces memory usage, particularly on low usage systems.